src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java

```diff
@@ -258,10 +257,11 @@ public void leaveToken(DetailAST ast) {
      */
     private void checkIfSemicolonIsInDifferentLineThanPrevious(DetailAST ast) {
         DetailAST currentStatement = ast;
-        final boolean hasResourcesPrevSibling =
-                currentStatement.getPreviousSibling() != null
-                        && currentStatement.getPreviousSibling().getType() == TokenTypes.RESOURCES;
-        if (!hasResourcesPrevSibling && isMultilineStatement(currentStatement)) {
+        final DetailAST previousSibling = ast.getPreviousSibling();
+        final boolean isUnnecessarySemicolon = previousSibling == null
+            || previousSibling.getType() == TokenTypes.RESOURCES
+            || ast.getParent().getType() == TokenTypes.COMPILATION_UNIT;
+        if (!isUnnecessarySemicolon) {
```

Comment:
Is this really unnecessary or more ignorable?

Reply:
It's an unnecessary semicolon.

Reply:
I am just confused then what is unnecessary mean to a check that checks if there is more than one statement per line. If 2 semicolons are on the same line, regardless of code between them, that is a flag for this check.
Are you sure we are not saying the semicolons can just be ignored in this case since it won't apply to this check?

Reply:
We still check this semicolon, so it's surely not ignorable, it is unnecessary as-

1. The `RESOURCES` part
```java
try (OutputStream s10 = new PipedOutputStream();       
     OutputStream s11 = new PipedOutputStream(); s2;) {
}                                                      
```
```
LITERAL_TRY -> try 
 |--RESOURCE_SPECIFICATION -> RESOURCE_SPECIFICATION 
 |   |--LPAREN -> ( 
 |   |--RESOURCES -> RESOURCES 
 |   |   |--RESOURCE -> RESOURCE 
 |   |   |   |--MODIFIERS -> MODIFIERS 
 |   |   |   |--TYPE -> TYPE 
 |   |   |   |   `--IDENT -> OutputStream 
 |   |   |   |--IDENT -> s10 
 |   |   |   `--ASSIGN -> = 
 |   |   |       `--EXPR -> EXPR 
 |   |   |           `--LITERAL_NEW -> new 
 |   |   |               |--IDENT -> PipedOutputStream 
 |   |   |               |--LPAREN -> ( 
 |   |   |               |--ELIST -> ELIST 
 |   |   |               `--RPAREN -> ) 
 |   |   |--SEMI -> ; 
 |   |   |--RESOURCE -> RESOURCE 
 |   |   |   |--MODIFIERS -> MODIFIERS 
 |   |   |   |--TYPE -> TYPE 
 |   |   |   |   `--IDENT -> OutputStream 
 |   |   |   |--IDENT -> s11 
 |   |   |   `--ASSIGN -> = 
 |   |   |       `--EXPR -> EXPR 
 |   |   |           `--LITERAL_NEW -> new 
 |   |   |               |--IDENT -> PipedOutputStream 
 |   |   |               |--LPAREN -> ( 
 |   |   |               |--ELIST -> ELIST 
 |   |   |               `--RPAREN -> ) 
 |   |   |--SEMI -> ; 
 |   |   `--RESOURCE -> RESOURCE 
 |   |       `--IDENT -> s2 
 |   |--SEMI -> ; ---> This is the semi after s2, it is unnecessary
 |   `--RPAREN -> ) 
 `--SLIST -> { 
     `--RCURLY -> } 
```
We still check the semi I highlighted, just we do not change the current statement to the previous sibling.
2. The `COMPILATION_UNIT` part-
```java
import java.io.IOException;
;
```
```
COMPILATION_UNIT -> COMPILATION_UNIT 
|--IMPORT -> import 
|   |--DOT -> . 
|   |   |--DOT -> . 
|   |   |   |--IDENT -> java 
|   |   |   `--IDENT -> io 
|   |   `--IDENT -> IOException 
|   `--SEMI -> ; 
|--SEMI -> ; ---> This semi is unnecessary
```
We still check this semicolon, if there is another semicolon we give a violation, we do not ignore it but as it is unnecessary we do not change `currentStatement` to the `previousSibling`.